### PR TITLE
[core] Add min and max pitch options

### PR DESCRIPTION
**Pull request by jmalanen**
Fixes: mapbox/mapbox-gl-native-team#101

### CODE REVIEW

**Comment by tobrun** [comment]
cc @julianrex

File: src/mbgl/map/transform_state.cpp [hunk 1]
```diff
@@ -340,6 +340,26 @@ double TransformState::getMaxZoom() const {
     return scaleZoom(max_scale);
 }
 
+void TransformState::setMinPitch(const double pitch_) {
+    if (pitch_ <= maxPitch) {
+        minPitch = util::clamp(pitch_, util::PITCH_MIN, maxPitch);
+    }
```
Comment (alexshalamov) [inline src/mbgl/map/transform_state.cpp]: should we report warning if pitch is less than util::PITCH_MIN?

File: src/mbgl/map/transform_state.cpp [hunk 2]
```diff
@@ -340,6 +340,26 @@ double TransformState::getMaxZoom() const {
...
+}
+
+double TransformState::getMinPitch() const {
+    return minPitch;
+}
+
+void TransformState::setMaxPitch(const double pitch_) {
+    if (pitch_ >= minPitch) {
+        maxPitch = util::clamp(pitch_, minPitch, util::PITCH_MAX);
+    }
```
Comment (alexshalamov) [inline src/mbgl/map/transform_state.cpp]: warning in 'else' branch?

File: src/mbgl/map/transform.cpp [hunk 3]
```diff
@@ -379,6 +379,16 @@ void Transform::setMaxZoom(const double maxZoom) {
     state.setMaxZoom(maxZoom);
 }
 
+void Transform::setMinPitch(const double minPitch) {
+    if (std::isnan(minPitch)) return;
+    state.setMinPitch(minPitch * util::DEG2RAD);
```
Comment (brunoabinader) [inline src/mbgl/map/transform.cpp]: Should we be checking against `util:: PITCH_MIN` here?

File: src/mbgl/map/transform.cpp [hunk 4]
```diff
@@ -379,6 +379,16 @@ void Transform::setMaxZoom(const double maxZoom) {
...
+}
+
+void Transform::setMaxPitch(const double maxPitch) {
+    if (std::isnan(maxPitch)) return;
+    state.setMaxPitch(maxPitch * util::DEG2RAD);
```
Comment (brunoabinader) [inline src/mbgl/map/transform.cpp]: Should we be checking against `util:: PITCH_MAX` here?

File: src/mbgl/map/transform_state.cpp (hunk 1 above)
Comment (jmalanen) [reply src/mbgl/map/transform_state.cpp]: Added warning

File: src/mbgl/map/transform_state.cpp (hunk 2 above)
Comment (jmalanen) [reply src/mbgl/map/transform_state.cpp]: Added warning

File: src/mbgl/map/transform.cpp (hunk 3 above)
Comment (jmalanen) [reply src/mbgl/map/transform.cpp]: Added warning

File: src/mbgl/map/transform.cpp (hunk 4 above)
Comment (jmalanen) [reply src/mbgl/map/transform.cpp]: Added warning

**Pull request merged**